File: app/src/main/java/com/shevelev/comics_viewer/common/dialogs/ZoomedPagePreviewDialog.java
package com.shevelev.comics_viewer.common.dialogs;

import android.app.Activity;
import android.app.AlertDialog;
import android.graphics.Bitmap;
import android.view.LayoutInflater;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.ImageView;

import com.shevelev.comics_viewer.R;
import com.shevelev.comics_viewer.common.helpers.BitmapsHelper;
import com.shevelev.comics_viewer.common.structs.Size;

public class ZoomedPagePreviewDialog
{
    private String _initName;

    private Button _okButton;

    private final Activity parentActivity;
    private final String fullPathToImageFile;
    private final Size dialogSize;

    private AlertDialog dialog;

    private Size imageSize;

    public ZoomedPagePreviewDialog(Activity parentActivity, String fullPathToImageFile, Size dialogSize)
    {
        this.parentActivity = parentActivity;
        this.fullPathToImageFile = fullPathToImageFile;
        this.dialogSize = dialogSize;
    }

    private AlertDialog create()
    {
        LayoutInflater inflater = parentActivity.getLayoutInflater();

        AlertDialog.Builder builder = new AlertDialog.Builder(parentActivity);

        FrameLayout view=(FrameLayout)inflater.inflate(R.layout.dialog_preview_zoomed, null);

        Bitmap image=createBitmap();
        ImageView pageImage=((ImageView)view.findViewById(R.id.preview_page_image));
        pageImage.setImageBitmap(image);

        ImageView closeIcon=((ImageView)view.findViewById(R.id.closeIcon));
        closeIcon.setImageResource(R.drawable.ic_clear_yellow_48dp);
        closeIcon.setOnClickListener(v -> dialog.dismiss());

        builder.setView(view);

        return builder.create();
    }

    private Bitmap createBitmap()
    {
        Bitmap source=BitmapsHelper.loadFromFile(fullPathToImageFile);
        Size sourceSize=new Size(source.getWidth(), source.getHeight());

        Size targetSize=dialogSize.inscribe(sourceSize);

        imageSize = targetSize;
        Bitmap bmp=BitmapsHelper.scale(source, targetSize);

        return bmp;
    }

    public void show()
    {
        dialog = create();
        dialog.show();

//        dialog.getWindow().setLayout(imageSize.getWidth(), imageSize.getHeight());          // resize dialog to image's size
    }
}

File: app/src/main/java/com/shevelev/comics_viewer/common/dialogs/ZoomedPagePreviewDialog.kt
package com.shevelev.comics_viewer.common.dialogs

import android.app.Activity
import android.app.AlertDialog
import android.graphics.Bitmap
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import android.widget.ImageView
import com.shevelev.comics_viewer.R
import com.shevelev.comics_viewer.common.helpers.BitmapsHelper.loadFromFile
import com.shevelev.comics_viewer.common.helpers.BitmapsHelper.scale
import com.shevelev.comics_viewer.common.structs.Size

class ZoomedPagePreviewDialog(private val parentActivity: Activity, private val fullPathToImageFile: String, private val dialogSize: Size) {
    private val _initName: String? = null
    private val _okButton: Button? = null
    private var dialog: AlertDialog? = null

    private var imageSize: Size? = null
    private fun create(): AlertDialog {
        val inflater = parentActivity.layoutInflater
        val builder = AlertDialog.Builder(parentActivity)
        val view = inflater.inflate(R.layout.dialog_preview_zoomed, null) as FrameLayout
        val image = createBitmap()
        val pageImage = view.findViewById<View>(R.id.preview_page_image) as ImageView
        pageImage.setImageBitmap(image)
        val closeIcon = view.findViewById<View>(R.id.closeIcon) as ImageView
        closeIcon.setImageResource(R.drawable.ic_clear_yellow_48dp)
        closeIcon.setOnClickListener { v: View? -> dialog!!.dismiss() }
        builder.setView(view)
        return builder.create()
    }

    private fun createBitmap(): Bitmap {
        val source = loadFromFile(fullPathToImageFile)
        val sourceSize = Size(source.width, source.height)
        val targetSize = dialogSize.inscribe(sourceSize)
        imageSize = targetSize
        return scale(source, targetSize)
    }

    fun show() {
        dialog = create()
        dialog!!.show()
    }
}